src/logger/advanced-outputs.ts

```
@@ -0,0 +1,621 @@
+      throw new Error(`File rename not allowed outside safe directories: ${oldPath} -> ${newPath}`);
+    }
+
+    try {
+      // SECURITY: Both paths have been validated and restricted to safe directories
+      fs.renameSync(safeOldPath, safeNewPath);
+    } catch (error) {
+      const errorMessage = error instanceof Error ? error.message : String(error);
+      throw new Error(`Failed to rename file ${oldPath} to ${newPath}: ${errorMessage}`);
+    }
+  }
+}
```

Biome (1.9.4)

[error] Avoid classes that contain only static members.

Prefer using simple functions instead of classes with only static members.

(lint/complexity/noStaticOnlyClass)